pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/models/DummyTableUpsertMetadataManager.java

```diff
@@ -72,6 +73,11 @@ public PartitionUpsertMetadataManager getOrCreatePartitionManager(int partitionI
   public void stop() {
   }
 
+  @Override
+  public Map<Integer, Long> getPartitionToPrimaryKeyCount() {
+    return null;
```

Comment:
if `null` is allowed, then can return null for RealtimeTableDataManager.getUpsertPartitionToPrimaryKeyCount() as well. And annotate the method with `@Nullable`